glacier_mapping/data/mask.py

```diff
@@ -58,9 +60,18 @@ def generate_masks(img_paths, shps_paths, output_base="mask", out_dir=None):
         mask = generate_mask(img.meta, shps)
         out_path = pathlib.Path(out_dir, f"{output_base}_{k:02}")
         np.save(str(out_path), mask)
+        # get borders
```

Comment:
I don't understand why we need new code for this. Can't we just clip the shapefiles to the borders in advance, and use that as input into the previous implementation?

I'm just very wary about changes in the code that will make it less generic to other applications.

Reply:
Clipping shp files to borders will create more false +ves. We can crop the original images in GEE though or in a separate step other than masking and slicing. 
The code still runs without borders if it's not supplied (it will just act as before). If that causes concerns about being generic we can still have a separate script for that outside of the library.

Reply:
Ah, I remember this discussion now. Didn't we agree to do this as a separate filtering step within the preprocessing script?

In my view, the entire process of removing the images that are outside of the borders is a preprocessing step, and should not referenced in the training code. The training code should operate on more or less abstract input / output pairs, without needing context dependent knowledge.

It's not enough to make sure the code works when not supplied borders. My concern is not about functionality, it's about clarity and maintainability.

Reply:
I think it being in generate_masks is part of the preprocessing step rather than the training code. But we can leave it out completely and have a code the crop the entire masks without using any preprocessing functionality from the library. it will be a script to run (it might take long because of the extra read). If that's easier to maintain, then it's easy fix.

Reply:
Ah you're completely right, I was just having trouble jumping across the contexts in this merge. For some reason I thought this was happening in train, but I'm happy with it happening in `generate_masks`. This part looks good to me, then.